compiler/damlc/daml-stdlib-src/DA/Internal/Desugar.daml

```diff
@@ -48,6 +50,15 @@ class HasInterfaceTypeRep i => Implements t i where
   toInterfaceContractId : ContractId t -> ContractId i
   fromInterfaceContractId : ContractId i -> Update (Optional (ContractId t))
 
+class HasIsInterfaceType t where
+  _isInterfaceType : proxy t -> Bool
```

Comment:
somehow this feels a bit like boolean blindness. Can you remind me why you preferred this over having the optional typerep in here?

Reply:
Also seems like it might be a bit faster.

Reply:
It was just to avoid overlap with `HasTemplateTypeRep`. We already have two classes that return typeReps and I'm a bit hesitant to add another.pp